src/frontend/src/optimizer/plan_node/generic/log_scan.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::rc::Rc;
+
+use educe::Educe;
+use pretty_xmlish::Pretty;
+use risingwave_common::catalog::{ColumnDesc, Field, Schema, TableDesc};
+use risingwave_common::util::column_index_mapping::ColIndexMapping;
+use risingwave_common::util::sort_util::ColumnOrder;
+
+use super::GenericPlanNode;
+use crate::catalog::ColumnId;
+use crate::expr::{ExprRewriter, ExprVisitor};
+use crate::optimizer::optimizer_context::OptimizerContextRef;
+use crate::optimizer::property::FunctionalDependencySet;
+
+#[derive(Debug, Clone, Educe)]
+#[educe(PartialEq, Eq, Hash)]
+pub struct LogScan {
+    pub table_name: String,
+    /// Include `output_col_idx` and `op_column`
+    pub output_col_idx: Vec<usize>,
+    /// Descriptor of the table
+    pub table_desc: Rc<TableDesc>,
+    /// Catalog of the op column
+    pub op_column: Rc<ColumnDesc>,
```

Comment:
No need to store this `op_column`, and no need to include op column id in the `output_col_idx`. 

The `ColumnDesc` contains the field name, type, and column id. The op field name and type is fixed. The mocked column id of op column is used only when you filter out the the newly added op column id in the `output_col_idx`. But it's  unnecessary to include the op column id in the `output_col_idx` if you just don't store this extra column id. The created `ColumnDesc` is only used to initialize a `Field` struct, but the column id is not used in the `Field`. Besides, for the field of the extra op column, you don't have to create a `ColumnDesc` for it and initialize the `Field` from the `ColumnDesc`. Just fill in the fixed data type and field name of op column for the `Field`.